launch_ros/examples/lifecycle_pub_sub_launch.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Launch a talker and a listener."""
+
+import os
+import sys
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..', 'launch'))  # noqa
+
+import launch
+import launch.actions
+import launch.events
+
+from launch_ros import get_default_launch_description
+import launch_ros.actions
+import launch_ros.events
+import launch_ros.events.lifecycle
+
+import lifecycle_msgs.msg
+
+
+def main(argv=sys.argv[1:]):
+    """Main."""
+    def _on_transition_event(context):
+        print('got transition_event {}'.format(context.locals.event))
+
+    ld = launch.LaunchDescription()
+
+    # Launch the talker node.
+    talker_node = launch_ros.actions.LifecycleNode(
+        node_name='talker',
+        package='lifecycle', node_executable='lifecycle_talker', output='screen')
+    ld.add_action(talker_node)
+    # When the talker node reaches the PRIMARY_STATE_ACTIVE state, start the listener.
+    ld.add_action(launch.actions.RegisterEventHandler(launch_ros.event_handlers.OnStateTransition(
+        target_lifecycle_node=talker_node, goal_state='active',
+        entities=[
+            launch.actions.LogInfo(
+                msg="lifecycle node 'talker' reached the 'active' state, launching 'listener'."),
+            launch_ros.actions.LifecycleNode(
+                node_name='listener',
+                package='lifecycle', node_executable='lifecycle_listener', output='screen'),
+        ],
+    )))
+
+    # When the talker reaches 'configured', make it take the 'activate' transition.
+    ld.add_action(launch.actions.RegisterEventHandler(launch_ros.event_handlers.OnStateTransition(
+        target_lifecycle_node=talker_node, goal_state='inactive',
+        entities=[
+            launch.actions.LogInfo(
+                msg="lifecycle node 'talker' reached the 'unconfigured' state, 'activating'."),
+            launch.actions.EmitEvent(event=launch_ros.events.lifecycle.ChangeState(
+                lifecycle_node_matcher=launch.events.process.matches_action(talker_node),
+                transition_id=lifecycle_msgs.msg.Transition.TRANSITION_ACTIVATE,
+            )),
+        ],
+    )))
+
+    # Make the talker node take the 'configure' transition.
```

Comment:
for readability of this file I recommend reversing the creation of these actions so we read about them in the order that they would occur

Reply:
The event handler must be registered before sending the event. Do you have a suggestion for how to reorder it that works?

Reply:
I am not sure if we are talking about the same scale. What I meant was add the action that creates the node; add the action that configures the node; add the action that activates the node; add the action that starts the listener. But I think it wasn't clear so you thought I was referring to this individual add_action call.

Reply:
Checkout 46bd3e1